src/useRouteQuery/useRouteQuery.ts

```diff
@@ -48,7 +48,11 @@ function factory(): () => UseRouteQuery {
     }
 
     const set = (key: string, value: LocationQueryValue | LocationQueryValue[]): void => {
-      query[key] = value
+      if (value?.length === 0) {
```

Comment:
It might not be obvious from reading this in Github but `value`'s type ends up as `string | string[] | undefined`